stormpath/express-stormpath
# Social Login via API

Fixes #112.

This PR adds support for social login from SPAs with for example the Angular SDK.

### Includes

* Support for login using `providerData` instead of username/password with the login API
* A new API route, `/oauth/providers`, that returns a list of all OAuth providers and their configurations
* Logouts with the `Accept` header set to `application/json` will respond with 200 instead of a redirect

### Notes

* Is the route `/oauth/providers` the best route for this purpose?

### How to verify

1. Setup a server with this branch
2. Verify the login, logout, etc. works with the AngularJS SDK: <https://github.com/stormpath/stormpath-sdk-angularjs/pull/48>
3. Run tests
